File: src/page1/Euler001.java
package page1;

import euler.Project;

/**
 * @author Obicere
 */
public class Euler001 extends Project {

    @Override
    public int number() {
        return 1;
    }

    @Override
    public void run() {
        int sum = 0;
        for (int i = 3; i < 1000; i++) {
            if (i % 3 == 0 || i % 5 == 0) {
                sum += i;
            }
        }
        System.out.println(sum);
    }

}

File: src/cpp/page1/Euler001.cpp
#include <iostream>
#include "../euler/Problem.h"

class Euler001 : public virtual Problem
{
public:

	void Euler001::run()
	{
		int sum = 0;
		int i = 3;

		while (i < 1000)
		{
			if (!(i % 3))
			{
				sum = sum + i;
			}
			else if (!(i % 5))
			{
				sum = sum + i;
			}
			++i;
		}
		std::cout << "Sum " << sum << std::endl;
	}
};